dateutil/utils.py

```diff
@@ -1,6 +1,44 @@
 # -*- coding: utf-8 -*-
 from __future__ import unicode_literals
 
+from datetime import datetime, time
+
+
+def today(tzinfo=None):
+    """
+    Returns a :py:class:`datetime` representing the current day at midnight
+
+    :param tzinfo:
+        The time zone to attach (also used to determine the current day).
+
+    :return:
+        A :py:class:`datetime.datetime` object representing the current day
+        at midnight.
+    """
+
+    dt = datetime.now(tzinfo)
+    return datetime.combine(dt.date(), time(0, tzinfo=tzinfo))
```

Comment:
+1 for `today(tzinfo)`, but I'm _very_ wary of implementing behavior that differs from same-named stdlib behavior.  btw:

```
>>> tz = pytz.timezone('US/Pacific')
>>> pd.Timestamp.today(tz)
Timestamp('2017-10-14 17:19:04.081315-0700', tz='US/Pacific')
```

Reply:
@jbrockmendel Sorry, I miscalculated what would be shown in the diff here, this discussion *should* be taking place on #474, so I've responded [here](https://github.com/dateutil/dateutil/pull/474/files#r144732646)